src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Dialog.kt

```diff
@@ -20,45 +20,140 @@ package io.github.yuutoproject.yuutobot.commands
+            .get()
+            .build()
+
+        httpClient.newCall(request).execute().use { response ->
+            if (response.code != 200) {
+                throw Exception("Failed to sync backgrounds and characters with API")
+            }
+            val json = DataObject.fromJson(IOUtil.readFully(IOUtil.getBody(response)))
+            return Pair(
+                json.getArray("backgrounds").toList() as List<String>,
+                json.getArray("characters").toList() as List<String>
+            )
+        }
+    }
+
+    private val bgAndChars = getBackgroundsAndCharacters()
+    private val backgrounds = bgAndChars.first
+    private val characters = bgAndChars.second
+
+    private val backgroundsString = "`${backgrounds.joinToString("`, `")}`"
+    private val charactersString = "`${characters.joinToString("`, `")}`"
+
+    private val json = "application/json; charset=utf-8".toMediaType()
 
-// TODO: get arch to implement the actual command
-class Dialog : AbstractCommand("dialog", CommandCategory.INFO, "does something", "[bg] <char> <text>") {
     override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
-        val moveThisToAStorageLocation = OkHttpClient()
+        val now = System.currentTimeMillis()
+
+        if (args.count() < 2) {
+            event.channel.sendMessage("This command requires two arguments : `dialog [background] <character> <text>` ([] is optional)").queue()
+            return
+        }
+
+        var character = args.removeAt(0).toLowerCase()
+        val background: String
+
+        if (characters.contains(character)) {
+            background = "camp"
+        } else {
+            background = character
+            character = args.removeAt(0).toLowerCase()
+        }
+
+        if (!backgrounds.contains(background)) {
+            event.channel.sendMessage("Sorry but I couldn't find `$background` as a location\nAvailable backgrounds are: $backgroundsString").queue()
+            return
+        }
+
+        if (!characters.contains(character)) {
+            event.channel.sendMessage("Sorry, but I don't think that `$character` is a character in Camp Buddy\nAvailable characters are: $charactersString").queue()
+            return
+        }
+
+        if (args.count() < 1) {
+            event.channel.sendMessage("Please provide a message to be written~!").queue()
+            return
+        }
+
+        val text = args.joinToString(" ").replace("/[‘’]/g".toRegex(), "'")
+
+        if (text.length > 140) {
+            event.channel.sendMessage("Sorry, the message limit is 140 characters <:hiroJey:692008426842226708>").queue()
+            return
+        }
+
+        if (
+            event.message.mentionedMembers.isNotEmpty() ||
+            EMOJI_REGEX.containsMatchIn(text) ||
+            NONASCII_REGEX.containsMatchIn(text)
+        ) {
+            event.channel.sendMessage("Sorry, I can't display emotes, mentions, or non-latin characters").queue()
```

Comment:
fix inconsistent error messages 
```suggestion
            event.channel.sendMessage("Sorry, but I can't display emotes, mentions, or non-latin characters").queue()
```